fdbclient/MultiVersionAssignmentVars.h

```diff
@@ -130,7 +130,10 @@ class DLThreadSingleAssignmentVar final : public ThreadSingleAssignmentVar<T> {
 	                            std::function<T(FdbCApi::FDBFuture*, FdbCApi*)> extractValue)
 	  : api(api), f(f), extractValue(extractValue), futureRefCount(1) {
 		ThreadSingleAssignmentVar<T>::addref();
-		api->futureSetCallback(f, &futureCallback, this);
+		FdbCApi::fdb_error_t e = api->futureSetCallback(f, &futureCallback, this);
+		if (e) {
+			throw Error(e);
```

Comment:
`DLThreadSingleAssignmentVar` is used in a function called `toThreadFuture` that unsurprisingly returns a `ThreadFuture`. It may make sense for us to modify that function to catch this error and pass it through the `ThreadFuture` rather than allow the exception to bubble out, what do you think?